translate_localization.py

```diff
@@ -8,49 +8,77 @@
+    token = token_from_file or os.getenv("CHATGPT_TOKEN")
+    context = context_from_file
 
-task_description = """
-I want you to translate some text from German to English.
-This text will be used to offer an iOS app in different languages.
-The input given to you will consist of three lines for each phrase that needs to be translated.
-First, the phrase in German.
-Second, a comment that describes in which context the phrase is occuring in the application's UI. Make sure that the translation you provide fits this context.
-Thrid, a line starting with "translation: " in which you should add your translation.
+    return token, context
 
-Please return only the lines starting with "translation:" with your added translation after the colon. Do not include the comments in the translations, those are only to add context.
-"""
+CHATGPT_TOKEN, APP_CONTEXT = get_config()
 
-additional_context_info = """
-In the data you will be given, the word "Zähler" refers to a digital power meter or an electricity meter. Do not translate it with "counter".
-"""
+if CHATGPT_TOKEN is None:
+    print("Usage: You have to set a CHATGPT_TOKEN environment var or provide a translate_info.py in the same folder with a CHATGPT_TOKEN constant.")
+    sys.exit(1)
 
+if len(sys.argv) < 2:
+    print("Usage: python translate_localization.py <language_code> [Localizable.xcstrings path]")
+    sys.exit(1)
+    
+target_language = sys.argv[1]  # The language code is the first argument
+print("Language code set to:", target_language)
+
+# Check if the path to Localizable.xcstrings is provided as the second argument
+if len(sys.argv) > 2:
+    localizable_file = sys.argv[2]
+else:
+    # Search for an Localizable.xcstrings file in the current directory and its subdirectories
+    localizables = glob.glob("**/Localizable.xcstrings", recursive=True)
+    if localizables:
+        localizable_file = localizables[0]  # Take the first Localizable.xcstrings file found
+    else:
+        print("Error: No Localizable.xcstrings found in the currenty directory and its subdirectories")
+        sys.exit(1)
+
+print("Using Localizable.xcstrings:", localizable_file)
 
 class Translatable:
 
     def __init__(self, key, info_dict):
         self.key = key
         self.info_dict = info_dict
     
-    def is_translated_in(self, language: str = "en"):
+    def is_translated_in(self, language: str = target_language):
         if "localizations" in self.info_dict:
             l10ns = self.info_dict["localizations"]
             return language in l10ns.keys()
         return False
     
     def get_gpt_query(self) -> str:
-        query = f"key: {self.key}\n"
+        query = f"key: {self.key}\\n"
         comment = self.info_dict["comment"] if "comment" in self.info_dict else "No comment provided."
-        query += f"comment: {comment}\n"
-        query += "translation: \n"
+        query += f"comment: {comment}\\n"
+        query += "translation: \\n"
```

Comment:
Why are you changing the newlines to escaped backslashes + n?

Reply:
Probably I mass replaced something and I made this on mistake, restoring it now